mavconn/mavconn.py

```diff
@@ -8,6 +14,18 @@ class MAVLinkConnection:
     def __init__(self, mavfile):
         self.mav = mavfile
         self._stacks = defaultdict(list)
+        self._timers = []
+
+    def __enter__:
+        condition = threading.Condition()
+        executor = ThreadPoolExecutor(max_workers=3) #change this later
+        timer_thread = executor.submit(timer_work, condition)
```

Comment:
Use regular threading.Thread (https://docs.python.org/3.7/library/threading.html) here.  ThreadPool can run out of threads, this is fine for handlers (and is even what you want) but should not happen for the timer or listener threads.  Those threads are application lifetime (or rather MAVLinkConnection) lifetime threads and thus should have their own threads, not shared with the pool.